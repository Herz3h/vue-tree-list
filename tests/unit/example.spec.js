import { mount } from "@vue/test-utils";
import { Tree, VueTreeList } from "@/index";

describe("VueTreeList", () => {
  it("renders correctly", () => {
    const wrapper = mount(VueTreeList, {
      propsData: {
        model: new Tree([
          {
            name: "Node 1",
            id: 1,
            pid: 0,
            dragDisabled: true,
            addTreeNodeDisabled: true,
            addLeafNodeDisabled: true,
            editNodeDisabled: true,
            delNodeDisabled: true,
            children: [
              {
                name: "Node 1-2",
                id: 2,
                isLeaf: true,
                pid: 1
              }
            ]
          },
          {
            name: "Node 2",
            id: 3,
            pid: 0,
            disabled: true
          },
          {
            name: "Node 3",
            id: 4,
            pid: 0
          }
        ]),
        defaultTreeNodeName: "new node",
        defaultLeafNodeName: "new leaf",
        defaultExpanded: false
      }
    });

    expect(wrapper.text()).toMatch(`Node 1················
            Node 1-2················
            Node 2················
            Node 3`);
  });
});
