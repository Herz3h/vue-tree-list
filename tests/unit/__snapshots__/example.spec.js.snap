// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueTreeList renders correctly 1`] = `
<div class="vtl">
  <!---->
  <div class="">
    <div class="vtl">
      <div>
        <div class="vtl-border vtl-up"></div>
        <div id="1" draggable="false" class="vtl-tree-node vtl-drag-disabled"><span class="vtl-caret vtl-is-small"><i class="vtl-icon vtl-icon-caret-right"></i></span> <span><i class="vtl-icon vtl-menu-icon vtl-icon-folder"></i></span>
          <div class="vtl-node-content">
            Node 1
          </div>
          <div class="vtl-operation" style="display: none;">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
        <!---->
      </div>
      <div class="vtl-tree-margin" style="display: none;">
        <div class="vtl">
          <div>
            <div class="vtl-border vtl-up"></div>
            <div id="2" draggable="true" class="vtl-tree-node">
              <!----> <span><i class="vtl-icon vtl-menu-icon vtl-icon-file"></i></span>
              <div class="vtl-node-content">
                Node 1-2
              </div>
              <div class="vtl-operation" style="display: none;">
                <!---->
                <!----> <span title="edit"><i class="vtl-icon vtl-icon-edit"></i></span> <span title="delete"><i class="vtl-icon vtl-icon-trash"></i></span></div>
            </div>
            <!---->
          </div>
          <!---->
        </div>
      </div>
    </div>
    <div class="vtl">
      <div>
        <div class="vtl-border vtl-up"></div>
        <div id="3" draggable="true" class="vtl-tree-node vtl-disabled">
          <!----> <span><i class="vtl-icon vtl-menu-icon vtl-icon-folder"></i></span>
          <div class="vtl-node-content">
            Node 2
          </div>
          <div class="vtl-operation" style="display: none;"><span title="add tree node"><i class="vtl-icon vtl-icon-folder-plus-e"></i></span> <span title="add leaf node"><i class="vtl-icon vtl-icon-plus"></i></span> <span title="edit"><i class="vtl-icon vtl-icon-edit"></i></span> <span title="delete"><i class="vtl-icon vtl-icon-trash"></i></span></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
    <div class="vtl">
      <div>
        <div class="vtl-border vtl-up"></div>
        <div id="4" draggable="true" class="vtl-tree-node">
          <!----> <span><i class="vtl-icon vtl-menu-icon vtl-icon-folder"></i></span>
          <div class="vtl-node-content">
            Node 3
          </div>
          <div class="vtl-operation" style="display: none;"><span title="add tree node"><i class="vtl-icon vtl-icon-folder-plus-e"></i></span> <span title="add leaf node"><i class="vtl-icon vtl-icon-plus"></i></span> <span title="edit"><i class="vtl-icon vtl-icon-edit"></i></span> <span title="delete"><i class="vtl-icon vtl-icon-trash"></i></span></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
  </div>
</div>
`;
